DiffusionPreprocessing/scripts/basic_preproc_best_b0.sh

```diff
@@ -0,0 +1,275 @@
+  # find index of minimum score
+  scores=()
+  while read line; do scores+=("$line"); done < ${rawdir}/select_b0_${pe_sign}/scores.txt
+  min_idx=0
+  for idx in $(seq 0 $((`${FSLDIR}/bin/fslval ${rawdir}/all_${pe_sign}_b0s dim4` - 1))) ; do
+    if [ $(echo "${scores[${idx}]} < ${scores[${min_idx}]}" | bc -l) -eq 1 ] ; then min_idx=$idx ; fi
+  done
+  echo "Selecting ${pe_sign} B0 with index ${min_idx}" 
+  echo "${pe_sign} ${min_idx}" >> ${rawdir}/index_best_b0s.txt
+  ${FSLDIR}/bin/fslroi ${rawdir}/all_${pe_sign}_b0s ${rawdir}/best_${pe_sign}_b0 ${min_idx} 1
+done
+
+# producing acqparams.txt
+if [ ${PEdir} -eq 1 ]; then    #RL/LR phase encoding
+  echo 1 0 0 ${ro_time} > ${rawdir}/acqparams.txt
+  echo -1 0 0 ${ro_time} >> ${rawdir}/acqparams.txt
+elif [ ${PEdir} -eq 2 ]; then  #AP/PA phase encoding
+  echo 0 1 0 ${ro_time} > ${rawdir}/acqparams.txt
+  echo 0 -1 0 ${ro_time} >> ${rawdir}/acqparams.txt
+fi
+
+################################################################################################
+## Merging Files and correct number of slices 
+################################################################################################
+echo "Merging Pos and Neg images"
+${FSLDIR}/bin/fslmerge -t ${rawdir}/Pos `echo ${rawdir}/${basePos}_[0-9]*.nii*`
+${FSLDIR}/bin/fslmerge -t ${rawdir}/Neg `echo ${rawdir}/${baseNeg}_[0-9]*.nii*`
+
+paste -d' ' `echo ${rawdir}/${basePos}*.bval` >${rawdir}/Pos.bval
+paste -d' ' `echo ${rawdir}/${basePos}*.bvec` >${rawdir}/Pos.bvec
+paste -d' ' `echo ${rawdir}/${baseNeg}*.bval` >${rawdir}/Neg.bval
+paste -d' ' `echo ${rawdir}/${baseNeg}*.bvec` >${rawdir}/Neg.bvec
+
+# start index file with a 1 to indicate the reference B0 image
+echo 1 > ${rawdir}/index.txt
+for idx in $(seq 1 `${FSLDIR}/bin/fslval ${rawdir}/Pos dim4`) ; do
+  echo 1 >> ${rawdir}/index.txt
+done
+for idx in $(seq 1 `${FSLDIR}/bin/fslval ${rawdir}/Neg dim4`) ; do
+  echo 2 >> ${rawdir}/index.txt
+done
+
+dimz=`${FSLDIR}/bin/fslval ${rawdir}/Pos dim3`
+if [ `isodd $dimz` -eq 1 ];then
+	echo "Remove one slice from data to get even number of slices"
+	${FSLDIR}/bin/fslroi ${rawdir}/Pos ${rawdir}/Posn 0 -1 0 -1 1 -1
+	${FSLDIR}/bin/fslroi ${rawdir}/Neg ${rawdir}/Negn 0 -1 0 -1 1 -1
+	${FSLDIR}/bin/fslroi ${rawdir}/best_Pos_b0 ${rawdir}/best_Pos_b0n 0 -1 0 -1 1 -1
+	${FSLDIR}/bin/fslroi ${rawdir}/best_Neg_b0 ${rawdir}/best_Neg_b0n 0 -1 0 -1 1 -1
+	${FSLDIR}/bin/imrm ${rawdir}/Pos
+	${FSLDIR}/bin/imrm ${rawdir}/Neg
+	${FSLDIR}/bin/imrm ${rawdir}/best_Pos_b0
+	${FSLDIR}/bin/imrm ${rawdir}/best_Neg_b0
+	${FSLDIR}/bin/immv ${rawdir}/Posn ${rawdir}/Pos
+	${FSLDIR}/bin/immv ${rawdir}/Negn ${rawdir}/Neg
+	${FSLDIR}/bin/immv ${rawdir}/best_Pos_b0n ${rawdir}/best_Pos_b0
+	${FSLDIR}/bin/immv ${rawdir}/best_Neg_b0n ${rawdir}/best_Neg_b0
+fi
+
+echo "Perform final merge"
+${FSLDIR}/bin/fslmerge -t ${rawdir}/Pos_Neg_b0 ${rawdir}/best_Pos_b0 ${rawdir}/best_Neg_b0
+# include Pos_b0 as the first volume of Pos_Neg, so that eddy will use it as reference
```

Comment:
```suggestion
# include Pos_b0 as the first volume of Pos_Neg, so that eddy will use it as reference
# it will be removed after eddy is completed
```